addon/PageScraper.js

```diff
@@ -30,33 +31,64 @@ function PageScraper(previewProvider, tabTracker, options = {}) {
 
 PageScraper.prototype = {
   /**
-   * Check if the link is already in the metadata database. If not, parse the
-   * HTML and insert it in the metadata database
+   * Parse the HTML and attempt to insert it in the metadata database
    */
   _asyncParseAndSave: Task.async(function*(rawHTML, url, event) {
-    let startTime = Date.now();
-    this._tabTracker.handlePerformanceEvent(event, "metadataReceivedRawHTML", startTime);
-
-    let link = yield this._previewProvider.asyncLinkExist(url);
-    if (!link) {
+    this._tabTracker.handlePerformanceEvent(event, "metadataReceivedRawHTML", Date.now());
```

Comment:
Move all the performance ping key names into a constant object

    const TELEMETRY_EVENTS = {
      RECEIVED_HTML: "metadataReceivedRawHTML",
      ...
    }